lib/assert.js

```diff
@@ -474,7 +474,13 @@ class Comparison {
 }
 
 function compareExceptionKey(actual, expected, key, message, keys) {
-  if (!(key in actual) || !isDeepStrictEqual(actual[key], expected[key])) {
+  let failed;
+  if (key in actual) {
+    failed = !isDeepStrictEqual(actual[key], expected[key]);
+  } else {
+    failed = expected[key] !== assert.undefinedProperty;
```

Comment:
It seems like this is is *not* about properties with the value `undefined`, but rather about *missing* properties… is that correct? If so, I think we should look for another name…

Reply:
That is correct. I would be happy about a better name but I could not find one.